src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -670,215 +768,169 @@ object OrderedRVD {
       out
     }.collect()
 
-    pkis.sortBy(_.min)(typ.pkType.ordering.toOrdering)
+    pkis.sortBy(_.min)(typ.kType.ordering.toOrdering)
   }
 
   def coerce(
     typ: OrderedRVDType,
     rvd: RVD
-  ): OrderedRVD = coerce(typ, rvd, None, None)
+  ): OrderedRVD = coerce(typ, rvd, None)
 
   def coerce(
     typ: OrderedRVDType,
     rvd: RVD,
     fastKeys: ContextRDD[RVDContext, RegionValue]
-  ): OrderedRVD = coerce(typ, rvd, Some(fastKeys), None)
+  ): OrderedRVD = coerce(typ, rvd, Some(fastKeys))
 
   def coerce(
     typ: OrderedRVDType,
     rvd: RVD,
-    hintPartitioner: OrderedRVDPartitioner
-  ): OrderedRVD = coerce(typ, rvd, None, Some(hintPartitioner))
-
-  def coerce(
-    typ: OrderedRVDType,
-    rvd: RVD,
-    fastKeys: Option[ContextRDD[RVDContext, RegionValue]],
-    hintPartitioner: Option[OrderedRVDPartitioner]
-  ): OrderedRVD = coerce(typ, rvd.crdd, fastKeys, hintPartitioner)
+    fastKeys: Option[ContextRDD[RVDContext, RegionValue]]
+  ): OrderedRVD = coerce(typ, rvd.crdd, fastKeys)
 
   def coerce(
     typ: OrderedRVDType,
     rdd: RDD[RegionValue]
-  ): OrderedRVD = coerce(typ, rdd, None, None)
+  ): OrderedRVD = coerce(typ, rdd, None)
 
   def coerce(
     typ: OrderedRVDType,
     rdd: RDD[RegionValue],
     fastKeys: RDD[RegionValue]
-  ): OrderedRVD = coerce(typ, rdd, Some(fastKeys), None)
+  ): OrderedRVD = coerce(typ, rdd, Some(fastKeys))
 
   def coerce(
     typ: OrderedRVDType,
     rdd: RDD[RegionValue],
-    hintPartitioner: OrderedRVDPartitioner
-  ): OrderedRVD = coerce(typ, rdd, None, Some(hintPartitioner))
-
-  def coerce(
-    typ: OrderedRVDType,
-    rdd: RDD[RegionValue],
-    fastKeys: RDD[RegionValue],
-    hintPartitioner: OrderedRVDPartitioner
-  ): OrderedRVD = coerce(
-    typ,
-    rdd,
-    Some(fastKeys),
-    Some(hintPartitioner))
-
-  def coerce(
-    typ: OrderedRVDType,
-    rdd: RDD[RegionValue],
-    fastKeys: Option[RDD[RegionValue]],
-    hintPartitioner: Option[OrderedRVDPartitioner]
+    fastKeys: Option[RDD[RegionValue]]
   ): OrderedRVD = coerce(
     typ,
     ContextRDD.weaken[RVDContext](rdd),
-    fastKeys.map(ContextRDD.weaken[RVDContext](_)),
-    hintPartitioner)
+    fastKeys.map(ContextRDD.weaken[RVDContext](_))
+    )
```

Comment:
I think we can just delete the three `RDD` methods now, they're not used, AFAICT.